api/letterboxd.ts

```diff
@@ -0,0 +1,103 @@
+import {Query} from './query'
+import {NoResultReason, Result} from './result'
+
+export async function fetchFilm(query: Query): Promise<Result> {
+	const base = 'https://watchlistpicker.com/api'
+	const userParam = `?users=${query.lists.join('&users=')}`
+	const intersectParam = (query.intersect) ? '&intersect=true' : ''
+	const ignoreParam = (query.ignore.length > 0)
+		? `&ignore=${query.ignore.join(',')}`
+		: ''
+
+	const apiUrl = `${base}${userParam}${intersectParam}${ignoreParam}`
+	return fetch(apiUrl)
+		.then(res => res.json())
+		.then(result => ({
+			status: 'success',
+			title: result['film_name'],
+			url: result['slug'],
+			img: result['image_url'],
+		}))
+}
+
+/**
+ * todo:
+ * 	- support actor/director lookups
+ * 	- support ignore options
+ * 	- support intersection
+ */
+export async function findFilm(query: Query): Promise<Result> {
+	if (query.lists.length == 0) {
+		return buildError(NoResultReason.UNKNOWN)
+	}
+
+	// note: this trick only works if unioning the lists together
+	const list = pickRandom(query.lists)
+
+	const letterboxdUrl = (looksLikeUserList(list))
+		? urlFromUserList(list)
+		: urlFromUsername(list)
+
+	try {
+		const response = await fetch(letterboxdUrl)
+
+		if (!response.ok) {
+			console.error(`Error fetching ${letterboxdUrl}: ${response.status} ${response.statusText}`)
+			return buildError(NoResultReason.UNKNOWN)
+		}
+
+		const html = await response.text()
+		// todo: parse?
+
+		return {
+			status: 'success',
+			title: html,
+			img:'https://a.ltrbxd.com/resized/sm/upload/8g/5p/p4/6b/8YWirGQidtZeSEmhqvQM5FrI6N1-0-230-0-345-crop.jpg?v=10e3695a0f',
+			url: letterboxdUrl.toString(),
```

Comment:
This is some testing garbage from my experiment doing the scraping stuff in TS. Because I hate Go. I was just loading the HTML into the title attribute to see what I was working with. As you can see from the todo, there's a lot more to fix before its an alternative. I've included it in here in case you wanted to build off of it